src/package.rs

```diff
@@ -13,78 +13,96 @@
 // limitations under the License.
 
 use std::{
-    collections::HashMap,
+    collections::BTreeMap,
+    env::current_dir,
     fmt::{self, Formatter},
-    fs::File,
     io::{self, Cursor, Read, Write},
     ops::Deref,
     path::{Path, PathBuf},
     str::FromStr,
-    sync::Arc,
 };
 
-use async_recursion::async_recursion;
 use bytes::{Buf, Bytes};
-use miette::{ensure, miette, Context, Diagnostic, IntoDiagnostic};
-use semver::{Version, VersionReq};
+use miette::{ensure, miette, Context, IntoDiagnostic};
+use semver::Version;
 use serde::{Deserialize, Serialize};
-use thiserror::Error;
 use tokio::fs;
 use walkdir::WalkDir;
 
 use crate::{
-    credentials::Credentials,
     errors::{DeserializationError, SerializationError},
-    lock::{LockedPackage, Lockfile},
-    manifest::{self, Dependency, Manifest, MANIFEST_FILE},
-    registry::{Artifactory, RegistryUri},
+    lock::LockedPackage,
+    manifest::{self, Manifest, MANIFEST_FILE},
+    registry::RegistryUri,
     ManagedFile,
 };
 
 /// IO abstraction layer over local `buffrs` package store
-pub struct PackageStore;
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PackageStore {
+    root: PathBuf,
```

Comment:
We had a similar PR by @uschi2000

See the comments of #24 

I have to say, i still dislike the oop complexity associated with this, i get that the imperative approach is not maintainable longterm – can we find a functional architecture?

Reply:
How do you define *OOP complexity*?

Can you give an example of how a functional architecture would look like here that would model this better?

Reply:
`struct` / _instance_ orientation rather than dataflow orientation

the former results in (potentially huge) interlinked objects, flying around with limited composability where as the latter typically separated data transformation (logic) from io and results in better composability, testability and decomp at the same time

Imo (these functions) where all, mostly unrelated from each other, as they are IO heavy and do very little data transformation. A functional design could be to separate the IO part completely

So in this case, as most of the functions do only IO, i dont see the need for them to "belong" to a struct, rather a module, and remove the functions that nest their IO behavior.

---

As i said on rob's pr:

We also dont really have a need currently for making the path configurable, if thats the case the change is valid, but until then it is unnecessary complexity (imo).